### PR TITLE
BUGFIX: Avoid null being used in trimExplode()

### LINKED ISSUE #1552: Argument 2 passed to Neos\Utility\Arrays::trimExplode() must be of the type string, null given
### Steps to Reproduce

1. Install Neos
2. Import Neos.Demo
3. Try to log in

#### Expected behavior

I see the Neos UI.

#### Actual behavior

`Argument 2 passed to Neos\Utility\Arrays::trimExplode() must be of the type string, null given, called in /…/Development/Cache/Code/Flow_Object_Classes/Neos_ContentRepository_Domain_Repository_NodeDataRepository.php on line 1068`

The passed `$nodeTypeFilter` can be `null`, according to the docblock. There is a test for it being an empty string, but `null` is actually not caught and causes the error.

### Affected Versions

Neos: master

Flow: master

Probably older versions, too, since the affected code is basically in place since 2014.

### PR DESCRIPTION
**Pull request by kdambekalns**
Fixes #1552

### CODE REVIEW

**Comment by kdambekalns** [comment]
Please check this, I'd put it onto 2.3 or later as needed. Basically I wonder why this happens now!?

**Comment by kitsunet** [comment]
For 4.0 I suggest we add type hints and just allow only strings for the filter...

**Comment by kdambekalns** [comment]
Interesting test failures…

**Comment by aertmann** [comment]
looks fine code wise

**Comment by kdambekalns** [comment]
So what I still don't get: why does all that fail now? What has changed?

**Comment by kitsunet** [comment]
That is an interesting question... I didn'T see any reason as well...

**Comment by kitsunet** [comment]
Pragmatic fix is something along the lines of:

`$nodeTypeFilter = empty($nodeTypeFilter) ? '' : $nodeTypeFilter;` inside
`\Neos\ContentRepository\Domain\Repository\NodeDataRepository::getNodeTypeFilterConstraintsForDql` 

and not checking anything before.

**Comment by kitsunet** [comment]
Ok, pushed my idea. Also I guess this happens because PHPs behavior changed for some reason. It seems a call of `$this->getNodeTypeFilterConstraintsForDql($variableBeingNullValue)` will not result in the default value of that argument anymore but in NULL.

File: Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php [hunk 1]
```diff
@@ -1058,8 +1058,9 @@ public function addNodeTypeFilterConstraintsToQueryBuilder(QueryBuilder $queryBu
      * @param string $nodeTypeFilter
      * @return array
      */
-    protected function getNodeTypeFilterConstraintsForDql($nodeTypeFilter)
+    protected function getNodeTypeFilterConstraintsForDql($nodeTypeFilter = '')
```
Comment (gerhard-boden) [inline Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php]: So this fallback value for the argument is never really used in PHP7? Strange (new) behaviour

**Comment by kdambekalns** [comment]
> will not result in the default value of that argument anymore but in NULL.

Well, _that has always been the case_. If you pass something, that is used. The default value is only ever used if the argument _is omitted entirely_. Nothing new or strange going on.

**Pull request merged**